clcache.py

```diff
@@ -89,6 +89,11 @@ def printBinary(rawData, stream):
         fp.write(rawData)
 
 
+def basenameWithoutExtension(path):
+    basename = os.path.basename(path)
+    return os.path.splitext(basename)[0]
+
+
```

Comment:
Seeing these use cases, how about a slightly more powerful function like

```python
def replaceInPath(path, extension=None, dirname=None):
    if extension is not None:
        path = os.path.splitext(path)[0] + extension
    if dirname is not None:
        path = os.path.join(dirname, os.path.basename(path))
    return path
```

...because you could then write...